src/utils/getUserRewards.ts

```diff
@@ -0,0 +1,201 @@
+import { BigNumber, BigNumberish, providers } from "ethers";
+import axios from "axios";
+import { maxBN, minBN, now, WAD } from "../helpers";
+import { GraphUserBalances, UserBalance, formatGraphBalances } from "./graph";
+import { Market } from "./graph/getGraphMarkets/markets.types";
+import { getEpochsBetweenTimestamps, timestampToEpoch } from "./timestampToEpoch";
+import { RewardsDistributor__factory } from "@morpho-labs/morpho-ethers-contract";
+import addresses from "@morpho-labs/morpho-ethers-contract/lib/addresses";
+import { ages } from "../ages";
+export const getUserRewards = async (
+  address: string,
+  blockNumber?: number,
+  provider: providers.Provider = new providers.InfuraProvider(1)
+) => {
+  let timestampEnd = now();
+  if (blockNumber) {
+    const block = await provider.getBlock(blockNumber);
+    timestampEnd = block.timestamp;
+  }
+  const userBalances = await getUserBalances(
+    ages[0].subgraphUrl, // TODO: export the subgraphUrl
+    address.toLowerCase(),
+    blockNumber
+  );
```

Comment:
Could be parallelized. Perhaps secondary, can you create an issue if you don't want to implement it now?